Механизм обработки исключений в игре "Космическая битва"

ms-pavlov opened the pull request:
Цель:
Научится писать различные стратегии обработки исключений так, чтобы соответствующий блок try-catсh не приходилось модифицировать каждый раз, когда возникает потребность в обработке исключительной ситуации по-новому.

- Реализовать Команду, которая повторяет Команду, выбросившую исключение. Реализовать Команду, которая записывает информацию о выброшенном исключении в лог.
- Реализовать обработчик исключения, который ставит команду, пишущую в лог, в очередь. Реализовать обработчик, который ставит в очередь команду повторитель.
- Реализовать стратегию обработки исключения - повторить два раза, потом записать в лог. Указание: создать новую команду, точно такую же как в пункте 3. Тип этой команды будет показывать, что Команду не удалось выполнить два раза.

ms-pavlov committed:
Механизм обработки исключений в игре "Космическая битва"

Цель:
Научится писать различные стратегии обработки исключений так, чтобы соответствующий блок try-catсh не приходилось модифицировать каждый раз, когда возникает потребность в обработке исключительной ситуации по-новому.

gudun-ku commented on src/main/java/ru/otus/architect/exceptionHandling/ExceptionHandlerService.java:
```diff
@@ -0,0 +1,17 @@
+import ru.otus.architect.commands.Command;
+import ru.otus.architect.exceptionHandling.handlers.CommandExceptionHandler;
+
+public interface ExceptionHandlerService {
```
тут мне кажется не совсем согласно SRP, все же handleException это дело хендлера, а регистрация правил это регистрация правил, но может я и ошибаюсь

gudun-ku commented on src/main/java/ru/otus/architect/exceptionHandling/handlers/CommandExceptionLogHandler.java:
```diff
@@ -0,0 +1,25 @@
+
+    @Override
+    public void handleException(Command command, Exception exception) {
+        if (!commandQueue.offer(new LogCommand(log, command, exception))) {
```
+ да, согласен, проверка нужная, очередь то может переполнится

gudun-ku commented on src/main/java/ru/otus/architect/commands/LogCommand.java:
```diff
@@ -0,0 +1,23 @@
+
+import ch.qos.logback.classic.Logger;
+
+public class LogCommand implements Command {
```
Хорошо реализовано

gudun-ku reviewed (approved):
Все очень хорошо, есть только пара комментов, по поводу повтора да, мне было интересно, я так и предполагал, но мне не супер понравилась идея )

ms-pavlov committed:
Механизм обработки исключений в игре "Космическая битва"

Цель:
Научится писать различные стратегии обработки исключений так, чтобы соответствующий блок try-catсh не приходилось модифицировать каждый раз, когда возникает потребность в обработке исключительной ситуации по-новому.

ms-pavlov committed:
Механизм обработки исключений в игре "Космическая битва"

Цель:
Научится писать различные стратегии обработки исключений так, чтобы соответствующий блок try-catсh не приходилось модифицировать каждый раз, когда возникает потребность в обработке исключительной ситуации по-новому.

ms-pavlov committed:
Механизм обработки исключений в игре "Космическая битва"

Цель:
Научится писать различные стратегии обработки исключений так, чтобы соответствующий блок try-catсh не приходилось модифицировать каждый раз, когда возникает потребность в обработке исключительной ситуации по-новому.

ms-pavlov committed:
Merge branch 'main' into hw03-ms-pavlov
